src/PDO/Builder.php

```diff
@@ -12,13 +12,18 @@ class Builder implements BuilderInterface
     protected const PROP_DATA_SOURCE_NAME = 'data_source_name';
     protected const PROP_USER_NAME        = 'user_name';
     protected const PROP_PASSWORD         = 'password';
+    protected const PROP_PORT             = 'port';
     protected const PROP_OPTIONS          = 'options';
     protected $_pdo;
 
     public function getPdo(): \PDO
     {
         if ($this->_pdo === null) {
-            $dsn = $this->_getDataSourceName();
+            if ($this->_getPort() !== 0){
```

Comment:
non-leading question - do we really want to use zero as an `uninitiated` value?

Reply:
I'm following the same patten we use for Redis:
```yaml
parameters:
  neighborhoods.kojo.environment.parameters.redis_port: 0
```

I want the getter and setter to be strictly typed to an `int`. Calling `setPort(null)` during DI would make require the removal of that strict type contract.

We cannot use `5432` or `3306` since the default depends on the database adapter. The `\PDO` object already handles picking the right default when omitting the `port=%d` from the DSN.